draft-ietf-sframe-enc.md

```diff
@@ -214,62 +222,71 @@ The E2EE keys used to encrypt the frame are exchanged out of band using a secure
 | |                                                       | |
 | |                                                       | |
 | |                                                       | |
-| |                  Encrypted Frame                      | |
+| |                   Encrypted Data                      | |
 | |                                                       | |
 | |                                                       | |
 | |                                                       | |
 | |                                                       | |
-+^+-------------------------------------------------------+^+
++>+-------------------------------------------------------+<+
 | |                 Authentication Tag                    | |
 | +-------------------------------------------------------+ |
 |                                                           |
 |                                                           |
-+----+Encrypted Portion            Authenticated Portion+---+
++---- Encrypted Portion            Authenticated Portion ---+
 ~~~~~
 
+When SFrame is applied per-packet, the payload of each packet will be an SFrame
+ciphertext.  When SFrame is applied per-frame, the SFrame ciphertext
+representing an encrypted frame will span several packets, with the header
+appearing in the first packet and the authentication tag in the last packet.
+
 ## SFrame Header
-Since each endpoint can send multiple media layers, each frame will have a unique frame counter that will be used to derive the encryption IV. The frame counter must be unique and monotonically increasing to avoid IV reuse.
 
-As each sender will use their own key for encryption, so the SFrame header will include the key id to allow the receiver to identify the key that needs to be used for decrypting.
+The SFrame header specifies two values from which encryption parameters are
+derived:
+
+* A Key ID (KID) that determines which encryption key should be used
+* A counter (CTR) that is used to construct the IV for the encryption
```

Comment:
That is stated just below here:

> Applications MUST ensure that each (KID, CTR) combination is used for exactly
one encryption operation.  Typically this is done by assigning each sender a KID
or set of KIDs, then having each sender use the CTR field to uniquely identify
each encryption operation.

We could add something at the end of that if you think it would clarify, like:

> Applications MUST ensure that each (KID, CTR) combination is used for exactly
one encryption operation.  Typically this is done by assigning each sender a KID
or set of KIDs, then having each sender use the CTR field to uniquely identify
each encryption operation, for example by populating the CTR field with a monotonically increasing counter that increments for each encryption operation.

Reply:
If any random CTR value would work as long as it is used a single time, I am fine with that. I would also state that having an incremental counter is the most optimum approach as the encoding of the CTR on the header has less overhead the smaller the CTR value is.